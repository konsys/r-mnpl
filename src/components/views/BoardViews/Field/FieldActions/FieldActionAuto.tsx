import { IFieldRent } from "../../../../../types/types";
import React from "react";

export const FieldActionAuto = (rent: IFieldRent) => {
  return (
    <>
      <div className="TableFieldcard-data-rows">
        <div>
          <div className="_title">1 полe</div>
          <div className="_value _type_money _base_rent">
            {rent && rent.baseRent}
          </div>
        </div>
        <div>
          <div className="_title">2 поля</div>
          <div className="_value _type_money _one_star_rent">
            {rent && rent.oneStar}
          </div>
        </div>
        <div>
          <div className="_title">3 поля</div>
          <div className="_value _type_money _two_star_rent">
            {rent && rent.twoStar}
          </div>
        </div>
        <div>
          <div className="_title">4 поля</div>
          <div className="_value _type_money _free_star_rent">
            {rent && rent.freeStar}
          </div>
        </div>
      </div>
    </>
  );
};
