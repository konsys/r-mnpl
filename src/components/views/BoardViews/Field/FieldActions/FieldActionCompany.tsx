import { IFieldRent } from "../../../../../types/types";
import React from "react";

export const FieldActionCompany = (rent: IFieldRent) => {
  return (
    <div className="TableFieldcard-data-rows">
      <div>
        <div className="_title">
          <span>Базовая рента</span>
        </div>
        <div className="_value _type_money _base_rent">
          {rent && rent.baseRent}
        </div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money _one_star_rent">
          {rent && rent.oneStar}
        </div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money _two_star_rent">
          {rent && rent.twoStar}
        </div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
          <span className="_star_small"></span>
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money _free_star_rent">
          {rent && rent.freeStar}
        </div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_small"></span>
          <span className="_star_small"></span>
          <span className="_star_small"></span>
          <span className="_star_small"></span>
        </div>
        <div className="_value _type_money _four_star_rent">
          {rent && rent.fourStar}
        </div>
      </div>
      <div>
        <div className="_title">
          <span className="_star_big"></span>
        </div>
        <div className="_value _type_money _big_star_rent">
          {rent && rent.bigStar}
        </div>
      </div>
    </div>
  );
};
