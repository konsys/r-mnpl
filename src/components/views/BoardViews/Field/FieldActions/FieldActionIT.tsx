import { IFieldRent } from "../../../../../types/types";
import React from "react";

export const FieldActionIT = (rent: IFieldRent) => {
  return (
    <>
      <div className="TableFieldcard-data-rows">
        <div>
          <div className="_title">1 полe</div>
          <div className="_value _type_x">
            <svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
              <path
                d="M97.33 51.18l-14.71 32.2a6.32 6.32 0 0 1-8.38 3.12l-32.2-14.7a6.32 6.32 0 0 1-3.12-8.38l14.7-32.2A6.32 6.32 0 0 1 62 28.08L94.2 42.8a6.32 6.32 0 0 1 3.13 8.38zM64.68 35.92a6 6 0 1 0-4.99 10.92 6 6 0 0 0 5-10.92zm5.94 15.91a6 6 0 1 0-5 10.93 6 6 0 0 0 5-10.92zm5.93 15.92a6 6 0 1 0-4.99 10.93 6 6 0 0 0 5-10.93zm-42.5-3.32a7.33 7.33 0 0 0 .05 6.15l-6.54 3.05a6.32 6.32 0 0 1-8.4-3.06L4.2 38.5a6.32 6.32 0 0 1 3.05-8.4l32.09-14.96a6.32 6.32 0 0 1 8.4 3.06l3.8 8.14c-.11.18-.24.35-.33.55L34.06 64.42zm-11.12-28.4a6 6 0 1 0-10.89 5.08 6 6 0 0 0 10.89-5.08zM25.1 54.9a6 6 0 1 0 5.07 10.88A6 6 0 0 0 25.1 54.9zm19.6-29.02a6 6 0 1 0-10.89 5.08 6 6 0 0 0 10.9-5.08z"
                n-clip-rule="evenodd"
                n-fill-rule="evenodd"
              ></path>
            </svg>
            × <span className="_mult_one">{rent && rent.baseRent}</span>
          </div>
        </div>
        <div>
          <div className="_title">2 поля</div>
          <div className="_value _type_x">
            <svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
              <path
                d="M97.33 51.18l-14.71 32.2a6.32 6.32 0 0 1-8.38 3.12l-32.2-14.7a6.32 6.32 0 0 1-3.12-8.38l14.7-32.2A6.32 6.32 0 0 1 62 28.08L94.2 42.8a6.32 6.32 0 0 1 3.13 8.38zM64.68 35.92a6 6 0 1 0-4.99 10.92 6 6 0 0 0 5-10.92zm5.94 15.91a6 6 0 1 0-5 10.93 6 6 0 0 0 5-10.92zm5.93 15.92a6 6 0 1 0-4.99 10.93 6 6 0 0 0 5-10.93zm-42.5-3.32a7.33 7.33 0 0 0 .05 6.15l-6.54 3.05a6.32 6.32 0 0 1-8.4-3.06L4.2 38.5a6.32 6.32 0 0 1 3.05-8.4l32.09-14.96a6.32 6.32 0 0 1 8.4 3.06l3.8 8.14c-.11.18-.24.35-.33.55L34.06 64.42zm-11.12-28.4a6 6 0 1 0-10.89 5.08 6 6 0 0 0 10.89-5.08zM25.1 54.9a6 6 0 1 0 5.07 10.88A6 6 0 0 0 25.1 54.9zm19.6-29.02a6 6 0 1 0-10.89 5.08 6 6 0 0 0 10.9-5.08z"
                n-clip-rule="evenodd"
                n-fill-rule="evenodd"
              ></path>
            </svg>
            × <span className="_mult_two">{rent && rent.oneStar}</span>
          </div>
        </div>
      </div>
    </>
  );
};
