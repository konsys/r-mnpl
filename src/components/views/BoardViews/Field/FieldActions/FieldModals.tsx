import React from "react";

export const FieldModals = () => <>FieldModals</>;
