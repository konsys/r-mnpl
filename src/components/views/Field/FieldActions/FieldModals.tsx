import React from "react";

export const FieldModals = () => <>1</>;
