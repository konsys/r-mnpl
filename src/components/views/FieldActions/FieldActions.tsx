import React from "react";

interface IFieldRent {
  baseRent: number;
  oneStar: number;
  twoStar: number;
  freeStar: number;
  fourStar: number;
  bigStar: number;
}

interface IFieldPrice {
  startPrice: number;
  pledgePrice: number;
  buyoutPrice: number;
  branchPrice: number;
}

interface IFieldModalPosition {
  top: number;
  left: number;
}

interface Prop {
  position: IFieldModalPosition;
  fieldGroup: number;
  name: string;
  groupName: string;
  rent: IFieldRent;
  price: IFieldPrice;
}

export const FieldActions = (prop: Prop) => (
  <div
    className="TableFieldcard"
    style={{ top: `${prop.position.top}px`, left: `${prop.position.left}px` }}
  >
    <div className="TableFieldcard-top _mnpl_color_3">
      <div className="TableFieldcard-top-main">Вконтакте</div>
      <div className="TableFieldcard-top-sub">Веб-сервисы</div>
    </div>
    <div className="_bg">
      <div className="TableFieldcard-buttons">
        <div className="_mortgage">Заложить</div>
      </div>
      <div className="TableFieldcard-data">
        <div className="TableFieldcard-data-rich">
          <div>Стройте филиалы, чтобы увеличить ренту.</div>
        </div>
        <div className="TableFieldcard-data-rows">
          <div>
            <div className="_title">
              <span>Базовая рента</span>
            </div>
            <div className="_value _type_money">100</div>
          </div>
          <div>
            <div className="_title">
              <span className="_star_small"></span>
            </div>
            <div className="_value _type_money">500</div>
          </div>
          <div>
            <div className="_title">
              <span className="_star_small"></span>
              <span className="_star_small"></span>
            </div>
            <div className="_value _type_money">1,500</div>
          </div>
          <div>
            <div className="_title">
              <span className="_star_small"></span>
              <span className="_star_small"></span>
              <span className="_star_small"></span>
            </div>
            <div className="_value _type_money">4,500</div>
          </div>
          <div>
            <div className="_title">
              <span className="_star_small"></span>
              <span className="_star_small"></span>
              <span className="_star_small"></span>
              <span className="_star_small"></span>
            </div>
            <div className="_value _type_money">6,250</div>
          </div>
          <div>
            <div className="_title">
              <span className="_star_big"></span>
            </div>
            <div className="_value _type_money">7,500</div>
          </div>
        </div>
        <div className="TableFieldcard-data-rows">
          <div>
            <div className="_title">Стоимость поля</div>
            <div className="_value _type_money">1,400</div>
          </div>
          <div>
            <div className="_title">Залог поля</div>
            <div className="_value _type_money">700</div>
          </div>
          <div>
            <div className="_title">Выкуп поля</div>
            <div className="_value _type_money">840</div>
          </div>
          <div>
            <div className="_title">Покупка филиала</div>
            <div className="_value _type_money">750</div>
          </div>
        </div>
      </div>
    </div>
  </div>
);
