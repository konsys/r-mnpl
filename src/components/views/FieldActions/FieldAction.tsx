import React from "react";
import { IFieldModalPosition, IField } from "../../../types/BoardTypes";
import { FieldActionCompany } from "./FieldActionCompany";

export interface IFieldAction extends IField {
  position: IFieldModalPosition;
  isActive: boolean;
}

export const FieldActions = ({
  fieldGroup,
  fieldGroupName,
  price,
  rent,
  position,
  isActive,
  name,
  description,
}: IFieldAction) => {
  console.log(1111111, rent);
  return (
    <div
      className="TableFieldcard"
      style={{
        top: `${position.top}px`,
        left: `${position.left}px`,
        display: isActive ? "block" : "none",
      }}
    >
      <div className={`TableFieldcard-top _mnpl_color_${fieldGroup}`}>
        <div className="TableFieldcard-top-main">{name}</div>
        <div className="TableFieldcard-top-sub">{fieldGroupName}</div>
      </div>
      <div className="_bg">
        <div className="TableFieldcard-buttons">
          <div className="_mortgage">Заложить</div>
        </div>
        <div className="TableFieldcard-data">
          <div className="TableFieldcard-data-rich">
            <div>{description}</div>
          </div>
          {rent && <FieldActionCompany {...rent} />}
          <div className="TableFieldcard-data-rows">
            <div>
              <div className="_title">Стоимость поля</div>
              <div className="_value _type_money">
                {price && price.startPrice}
              </div>
            </div>
            <div>
              <div className="_title">Залог поля</div>
              <div className="_value _type_money">
                {price && price.pledgePrice}
              </div>
            </div>
            <div>
              <div className="_title">Выкуп поля</div>
              <div className="_value _type_money">
                {price && price.buyoutPrice}
              </div>
            </div>
            <div>
              <div className="_title">Покупка филиала</div>
              <div className="_value _type_money">
                {price && price.branchPrice}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
