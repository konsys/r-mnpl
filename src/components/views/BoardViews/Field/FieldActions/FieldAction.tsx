import {
  FieldType,
  IField,
  IFieldAction,
  IFieldModalPosition,
  OutcomeMessageType,
} from "../../../../../types/types";

import { FieldActionAuto } from "./FieldActionAuto";
import { FieldActionCompany } from "./FieldActionCompany";
import { FieldActionIT } from "./FieldActionIT";
import React from "react";
import { closeFieldActionEvent } from "../../../../../stores/Board/FieldsStore";
import { getActingPlayer } from "../../../../../utils/players.utils";
import { sendBoardAction } from "stores/Board/ActionStore";

export interface IFieldModal extends IField {
  position: IFieldModalPosition;
  isActive: boolean;
}

export const FieldActions = ({
  fieldGroup,
  fieldGroupName,
  price,
  rent,
  position,
  isActive,
  name,
  description,
  type,
  status,
  fieldId,
}: IFieldModal) => {
  const player = getActingPlayer();
  return (
    <div
      className="TableFieldcard"
      style={{
        top: `${position.top}px`,
        left: `${position.left}px`,
        display: isActive ? "block" : "none",
      }}
    >
      <div className={`TableFieldcard-top _mnpl_color_${fieldGroup}`}>
        <div className="TableFieldcard-top-main">{name}</div>
        <div className="TableFieldcard-top-sub">{fieldGroupName}</div>
      </div>
      <div className="_bg">
        <div className="TableFieldcard-buttons">
          {status &&
            player?.userId === status?.userId &&
            status.fieldActions &&
            status.fieldActions.indexOf(IFieldAction.MORTGAGE) > -1 && (
              <div
                className="_mortgage"
                onClick={() => {
                  sendBoardAction({
                    action: OutcomeMessageType.OUTCOME_MORTGAGE_FIELD_CLICKED,
                    fieldId,
                  });
                  closeFieldActionEvent();
                }}
              >
                Заложить
              </div>
            )}
          {status &&
            player?.userId === status?.userId &&
            status.fieldActions &&
            status.fieldActions.indexOf(IFieldAction.UNMORTGAGE) > -1 && (
              <div
                className="_unmortgage"
                onClick={() => {
                  sendBoardAction({
                    action:
                      OutcomeMessageType.OUTCOME_UN_MORTGAGE_FIELD_CLICKED,
                    fieldId,
                  });
                  closeFieldActionEvent();
                }}
              >
                Выкупить
              </div>
            )}
          {status &&
            player?.userId === status?.userId &&
            status.fieldActions &&
            status.fieldActions.indexOf(IFieldAction.LEVEL_UP) > -1 && (
              <div
                className="_level_up"
                onClick={() => {
                  sendBoardAction({
                    action: OutcomeMessageType.OUTCOME_LEVEL_UP_FIELD_CLICKED,
                    fieldId,
                  });
                  closeFieldActionEvent();
                }}
              >
                Построить
              </div>
            )}
          {status &&
            player?.userId === status?.userId &&
            status.fieldActions &&
            status.fieldActions.indexOf(IFieldAction.LEVEL_DOWN) > -1 && (
              <div
                className="_level_down"
                onClick={() => {
                  sendBoardAction({
                    action: OutcomeMessageType.OUTCOME_LEVEL_DOWN_FIELD_CLICKED,
                    fieldId,
                  });
                  closeFieldActionEvent();
                }}
              >
                Продать
              </div>
            )}
        </div>
        <div className="TableFieldcard-data">
          <div className="TableFieldcard-data-rich">
            <div>{description}</div>
          </div>
          {(rent && type === FieldType.COMPANY && (
            <FieldActionCompany {...rent} />
          )) ||
            (rent && type === FieldType.AUTO && (
              <FieldActionAuto {...rent} />
            )) ||
            (rent && type === FieldType.IT && <FieldActionIT {...rent} />)}
          <div className="TableFieldcard-data-rows">
            <div>
              <div className="_title">Стоимость поля</div>
              <div className="_value _type_money">
                {price && price.startPrice}
              </div>
            </div>
            <div>
              <div className="_title">Залог поля</div>
              <div className="_value _type_money">
                {price && price.pledgePrice}
              </div>
            </div>
            <div>
              <div className="_title">Выкуп поля</div>
              <div className="_value _type_money">
                {price && price.buyoutPrice}
              </div>
            </div>
            {price && price.branchPrice > 0 && (
              <div>
                <div className="_title">Покупка филиала</div>
                <div className="_value _type_money">
                  {price && price.branchPrice}
                </div>
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};
