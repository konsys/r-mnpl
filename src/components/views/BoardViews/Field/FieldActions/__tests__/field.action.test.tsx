import { FieldActions } from "../FieldActions";
import React from "react";
import { shallow } from "enzyme";
import { testFieldActions } from "testMocks/field.actions";

describe("Field action test", () => {
  it("should render", () => {
    expect(shallow(<FieldActions {...testFieldActions} />)).toMatchSnapshot();
  });
});
