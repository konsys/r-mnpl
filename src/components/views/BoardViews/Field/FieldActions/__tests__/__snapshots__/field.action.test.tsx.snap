// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field action test should render 1`] = `ShallowWrapper {}`;
